pwa/src/service-worker/service-worker.js

```diff
@@ -57,45 +57,79 @@ export function setUpServiceWorker() {
         // Precache all of the assets generated by your build process.
         // Their URLs are injected into the manifest variable below.
         // This variable must be present somewhere in your service worker file,
-        // even if you decide not to use precaching. See https://cra.link/PWA
-        precacheAndRoute(self.__WB_MANIFEST || [])
+        // even if you decide not to use precaching. See https://cra.link/PWA.
+        // Includes all built assets and index.html
+        const precacheManifest = self.__WB_MANIFEST || []
+
+        // Extract index.html from the manifest to precache, then route
+        // in a custom way
+        const indexHtmlManifestEntry = precacheManifest.find(({ url }) =>
+            url.includes('index.html')
```

Comment:
Maybe this should be `url.endsWith('index.html')` too since this would match e.g. `index.html.js`

Reply:
Hah yeah, fair point 🙂